core/math/math_funcs.h

```diff
@@ -192,7 +192,10 @@ class Math {
 
 	static _ALWAYS_INLINE_ double abs(double g) { return absd(g); }
 	static _ALWAYS_INLINE_ float abs(float g) { return absf(g); }
-	static _ALWAYS_INLINE_ int abs(int g) { return g > 0 ? g : -g; }
+	static _ALWAYS_INLINE_ int8_t abs(int8_t g) { return g > 0 ? g : -g; }
+	static _ALWAYS_INLINE_ int16_t abs(int16_t g) { return g > 0 ? g : -g; }
+	static _ALWAYS_INLINE_ int32_t abs(int32_t g) { return g > 0 ? g : -g; }
+	static _ALWAYS_INLINE_ int64_t abs(int64_t g) { return g > 0 ? g : -g; }
```

Comment:
Could make use of optimized c functions here that the compiler understands:

```suggestion
	static _ALWAYS_INLINE_ int8_t abs(int8_t g) { return g > 0 ? g : -g; }
	static _ALWAYS_INLINE_ int16_t abs(int16_t g) { return g > 0 ? g : -g; }
	static _ALWAYS_INLINE_ int32_t abs(int32_t g) { return ::abs(g); }
	static _ALWAYS_INLINE_ int64_t abs(int64_t g) { return ::labs(g); }
```

Reply:
Is there any way to make compilation fail if a 128bit int is used somewhere? (which I assume causes implicit conversion)

Reply:
It would be possible to add the function and implement as `static_assert(false)`. SFINAE can't help us here i think, but it would be overkill anyway.

Reply:
You must mean 64bits, no (long long is almost walways 64 bits) ? There is no direct support for 128 bit integers in any CPU I can think of.